homeassistant/components/splunk/__init__.py

```diff
@@ -51,10 +51,9 @@
 def post_request(event_collector, body, headers, verify_ssl):
     """Post request to Splunk."""
     try:
-        payload = {"host": event_collector, "event": body}
         requests.post(
             event_collector,
-            data=json.dumps(payload, cls=JSONEncoder),
+            data=json.dumps(body, cls=JSONEncoder),
```

Comment:
Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices